### DISCUSSION TITLE
stopping/pausing AWS SQS listener

### DISCUSSION THREAD

**Discussion by springnirps** [awspring/spring-cloud-aws#455] [Q&A]
I'm using spring cloud aws messaging dependency.

I have a @SQSListerner that is consume messages. I need to be able to stop consuming of messages from SQS if I detect my backend is unavailable to process the messages.

I've tried autowiring SimpleMessageListenerContainer and calling the stop method but this is producing a timeout exception. I am probably using this wrong. What is best practice to stop/pause consuming of SQS message to where I can later start consuming them again when my backend is available again.

Thanks.
-psv

**Comment by maciejwalkowiak** [comment]
Calling `stop` on `SimpleMessageListenerContainer` should be the way to go. Can you check what value is set for `SimpleMessageListenerContainer#queueStopTimeout`?

**Reply by nekperu15739** [reply]
Hi, by default its 10000 milliseconds
